### PR TITLE
fix(config): use cspell-lib to handle findConfig function

### LINKED ISSUE #13: rspell won't work if installed globally with pnpm
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the bug

If you install the package using `pnpm i -g rspell`, the project won't run correctly and will display this error:

```shell
Error: Cannot find module '../node_modules/cspell/dist/util/fileHelper'
Require stack:
- C:\Users\<user>\AppData\Local\pnpm\global\5\.pnpm\rspell@1.6.2-1\node_modules\rspell\dist\cli.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (C:\Users\<user>\AppData\Local\pnpm\global\5\.pnpm\rspell@1.6.2-1\node_modules\rspell\dist\cli.js:2:604)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'C:\\Users\\<user>\\AppData\\Local\\pnpm\\global\\5\\.pnpm\\rspell@1.6.2-1\\node_modules\\rspell\\dist\\cli.js'
  ]
}
```

### Steps To Reproduce

Run `pnpm i -g rspell` on your shell.

### Screenshots

_No response_

### Additional context

This is probably thanks to pnpm caching design where `cspell` package isn't exposed to the actual project, instead using symlinks to pnpm's virtual store.

### PR DESCRIPTION
**Pull request by guilherssousa**
This PR closes #13.

By analyzing the source code, the actual findConfig responsability is to find a path to the specified config file from CLI arguments.  So we can just reimplement this logic part from the original `cspell-lib`s `readConfig` function.

By adding `cspell-lib` as a direct dependency, package managers with caching store strategies like `pnpm` now can interact correctly with the CLI.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,65 +1,66 @@
 {
-	"name": "rspell",
-	"version": "1.6.2",
```
Comment (guilherssousa) [inline package.json:3]: `.editorconfig` doesn't cover `.json` files 😢

Comment (Lioness100) [reply package.json:3]: Ahh! Sorry about that! I'll try to fix that when I get home.

File: src/config.ts [hunk 2]
```diff
@@ -1,21 +1,18 @@
...
 // eslint-disable-next-line import/no-relative-packages
-import { readConfig } from '../node_modules/cspell/dist/util/fileHelper';
 import { previousState } from './shared';
 
 let configPath: string | undefined;
 
 export const findConfig = async (config?: string) => {
 	// Try to locate a config file in the current working directory, or with the `config` option if it was provided.
-	const configInfo = await readConfig(config, process.cwd());
+	const configSource = config ?? (await searchForConfig(process.cwd()))?.__importRef?.filename;
```
Comment (Lioness100) [inline src/config.ts:12]: `searchForConfig` might reject the promise if nothing is found. If so, you should do something along the lines of: `config ?? (await searchForConfig(process.cwd()).then((res) => res.__importRef?.filename).catch(() => null)`

**Review by Lioness100 — COMMENTED** [review]
This looks great! I'll fix the editorconfig when I get home and then we can fix the formatting. Please take a look at my comment, and also remove the "external: ["cspell"]" (or something like that) in `tsup.config.ts`, as that's not needed anymore. It was put there so that, even though the `fileHelper.ts` was imported relatively, it wouldn't be bundled with the main code.

File: src/config.ts (hunk 2 above)
Comment (Lioness100) [reply src/config.ts:12]: Nevermind, it doesn't, ignore me haha

**Comment by guilherssousa** [comment]
Done, removed cspell from `tsup.config.ts`

**Comment by Lioness100** [comment]
Great, thanks! I added json to the editorconfig, could you reformat your changes?

**Comment by guilherssousa** [comment]
> Great, thanks! I added json to the editorconfig, could you reformat your changes?

Done :)

**Comment by Lioness100** [comment]
tsconfig.json and tsconfig.eslint.json too please <3. Also, did you test this? It looks like it will work great, but would be good to make sure.

**Comment by guilherssousa** [comment]
I ran `yarn test` after changes and the tests passed correctly, but good to notice:

I found that (at least in Windows) after installing rspell and running it for the first time, the code won't find `cspell.json` on default configuration path and it will throw an error. I'll write some tests to check this, but I'll be interesting to work on this specific problem on a different issue!

**Comment by Lioness100** [comment]
Oh, really? You bring up a good point, that the cli should probably create a config for the default path if none is found there, otherwise the whole default path thing would be useless. But I thought I caught all the potential errors if the file doesn't exist haha. If you want to look into that, that would be great! thank you so so so much again, I'm really happy someone is interested in this project <3

As for the testing thing, the tests are kind of outdated, so they only cover the actual typo fixing functionality and don't interact with config stuff. So I think just creating a cspell.json in the project directory, running `rspell`, using "ignore all" on a word and seeing if it appears there would be a good manual test.

**Comment by guilherssousa** [comment]
So, what do you think about just leaving this PR by adding a single passing test and working on this other on a different PR/Issue? So we can separate the topics better

```js
describe('findConfig', () => {
	test("should return the input if it's a valid path", async () => {
		const input = 'local_cspell.json';

		const config = await findConfig(input);

		expect(config).toBe(input);
	});
});

```

**Comment by guilherssousa** [comment]
New test added and `.json` files formatted

**Comment by Lioness100** [comment]
I don't think that test is very productive because the function just returns the path if there's one specifically inputted. I think the best thing to do would be to mock whatever fs api that cspell-lib uses to look at files, make it think there's a cspell.json in the cwd, and see if `findConfig` returns that path. But I'm fine with merging this with no tests as long as you can quickly test that yourself manually, and then we can revisit it in a different pr or something like that.

**Comment by Lioness100** [comment]
Thanks for the contribution!!

**Pull request merged**